st/clitests/fault_tolerence_spec.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/python3.6
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import sys
+import os
+import yaml
+import shutil
+from framework import Config
+from ldap_setup import LdapInfo
+from framework import S3PyCliTest
+from s3client_config import S3ClientConfig
+from s3fi import S3fiTest
+from awss3api import AwsTest
+import s3kvs
+import time
+import hashlib
+
+def file_md5(fname):
+    hash_md5 = hashlib.md5()
+    with open(fname, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            hash_md5.update(chunk)
+    return hash_md5.hexdigest()
+
+# Run before all to setup the test environment.
+def before_all():
+    print("Configuring LDAP")
+    S3PyCliTest('Before_all').before_all()
+
+# Run before all to setup the test environment.
+before_all()
+
+S3ClientConfig.pathstyle = False
+S3ClientConfig.ldapuser = 'sgiamadmin'
+S3ClientConfig.ldappasswd = LdapInfo.get_ldap_admin_pwd()
+Config.no_ssl = True
+Config.log_enabled = False
+
+# Set access key, secret key of s3_test account
+S3ClientConfig.access_key_id = 'AKIAJPINPFRBTPAYOGNA'
+S3ClientConfig.secret_key = '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'
+
+# Clear all S3 index data
+s3kvs.clean_all_data()
+s3kvs.create_s3root_index()
+
+# ********** Create 's3faultbucket' bucket in existing s3_test account********
+AwsTest('Create Bucket "s3faultbucket" using s3_test account')\
+    .create_bucket("s3faultbucket").execute_test().command_is_successful()
+
+# Enable S3 Fault tolerence mode by setting S3_MAX_EXTENDED_OBJECTS_IN_FAULT_MODE to non-zero value
+# TODO: 
+# 1. Change /opt/seagate/cortx/s3/conf/s3config.yaml
+# 2. Re-start S3 server or force S3 server to reload new config changes
+
+# Enable S3 FI to force object write failure to allow creation of fragments
+S3fiTest('s3cmd enable FI motr_obj_write_fail').enable_fi_offnonm("enable", "motr_obj_write_fail", "1", "1").execute_test().command_is_successful()
+
+# Upload object (of size 4.7MB) to bucket. 
```

Comment:
you can extend it further later to have 
1) upload of object size 1MB
2) upload of 5K object (here motr unit size is 4k)
2) upload of file of 0 size
3) FI set to say offset 3 for 4MB file